### PR TITLE
Ticket 6574 GUI add a refresh button

### PR DESCRIPTION
**Pull request by Adam-Szw**
### Description of work

Added a button to reload PV connection to the left side menu. This is done using the same mechanism as switching
to current instrument (literally by telling the GUI to do that but skipping console-killing step)

### Ticket

[*Link to Ticket*](https://github.com/ISISComputingGroup/IBEX/issues/6574)

### Acceptance criteria

There is a mechanism for refreshing the GUI, which remakes all the PV connections (like switching to the current instrument) but does not kill running scripts

### Unit tests

*Give an overview of unit tests you have added or modified, if applicable. The aim is provide information to help the reviewer*

### System tests

*Mention any automated tests or manual tests that you have added or modified, if applicable. The aim is provide information to help the reviewer*

### Documentation
*Highlight and provide a link to any additions or changes to the documentation, if applicable. The aim is provide information to help the reviewer*

---

#### Code Review

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?
- [ ] Is there associated PR for the [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)?

### Final Steps
- [ ] Reviewer has also merged the [release notes](https://github.com/ISISComputingGroup/IBEX/blob/master/release_notes/ReleaseNotes_Upcoming.md)

### CODE REVIEW

**Review by rerpha — COMMENTED** [review]
Looks good and works well, there are some minor changes I'd like to make, though I can make them if you think they are a good idea: 
- Instead of "Reset PVs" we should maybe call it "Refresh PVs" - resetting could give the impression that they will all be set to 0 
- We should have a nice tooltip like we do for the "Reset layout" 
I also noticed that there are quite a few
`java.lang.Error: SWT Resource was not properly disposed` errors in the logs, not sure if this is a general issue with switching instrument or not though. If i hit the button a few times I can get the GUI to freeze up as well which may be related.

**Comment by rerpha** [comment]
> @rerpha This error `java.lang.Error: SWT Resource was not properly disposed` is unrelated to this change. I thought it could've come from this: https://github.com/ISISComputingGroup/ibex_gui/pull/1398/files but it was actually happening before just less frequently (when switching configs only once and now a couple of times) - confirmed by checking out [6afac9e](https://github.com/ISISComputingGroup/ibex_gui/commit/6afac9edf93cb45b5e6f23c6436d7e3757888696) of master (commit before my groups panel changes). I will nonetheless fix it and make it a flash review.

Good idea, I did actually try to cause the freezing again earlier and couldn't reproduce, but it'd be good to fix it in a flash review as you said.

**Comment by Adam-Szw** [comment]
Okay - this should fix all the issues. 'SWT resource' problem is included. Do let me know what you think of renaming things and also if this solution to button spaming is good.

File: base/uk.ac.stfc.isis.ibex.epics.switching/src/uk/ac/stfc/isis/ibex/epics/switching/InstrumentSwitchers.java [hunk 1]
```diff
@@ -88,9 +98,18 @@ public Switcher getObservableSwitcher(OnInstrumentSwitch switchType) {
...
         instance.nothingSwitcher.switchInstrument(instrument);
         instance.closingSwitcher.switchInstrument(instrument);
         instance.observablePrefixChangingSwitcher.switchInstrument(instrument);
         instance.writablePrefixChangingSwitcher.switchInstrument(instrument);
+        // This is to prevent instrument from being switched too often
+        ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+        executorService.schedule(new Runnable() {
+			@Override
+			public void run() {
+		        switching = false;
```
Comment (rerpha) [inline base/uk.ac.stfc.isis.ibex.epics.switching/src/uk/ac/stfc/isis/ibex/epics/switching/InstrumentSwitchers.java:111]: This is more of a question than a comment, but how/where does this get set back to `true`?

**Review by rerpha — APPROVED** [review]

**Pull request merged**